src/dst/src/Distributed.cc

```diff
@@ -104,6 +106,32 @@ void Distributed::runLoadBalancer(const char* ip, unsigned short port)
   }
 }
 
+bool Distributed::lookUpWorkers(const char* DNS, unsigned short port)
+{
+  asio::io_service ios;
+  asio::ip::udp::resolver::query resolver_query(DNS, std::to_string(port), asio::ip::udp::resolver::query::numeric_service);
+  asio::ip::udp::resolver resolver(ios);
+  auto it = resolver.resolve(resolver_query);
+
+  int workers_count = 0;
+  asio::ip::udp::resolver::iterator it_end;
+
+  for (; it != it_end; ++it) {
+      asio::ip::udp::endpoint ep = it->endpoint();
+      addWorkerAddress(ep.address().to_string().c_str(), port);
+      workers_count += 1;
+    }
+
+  if(workers_count == 0){
+    logger_->error(utl::DST, 201, "Discovered zero workers with the given DNS.");
```

Comment:
firing an error might be an overkill. There might not be workers up for now but they might be up later. That's why I think there should be a discovery thread that's constantly looking up workers (every n seconds). What do you think?

Reply:
will reduce it to warn. Agreed with the discovery thread.